matches/dao/match-dao.go

```diff
@@ -0,0 +1,172 @@
+package dao
+
+import (
+	// pq acts as the driver for SQL requests
+	"database/sql"
+	"errors"
+
+	_ "github.com/lib/pq"
+)
+
+const connStr = "user=postgres dbname=postgres host=matches-db sslmode=disable"
+
+// MatchGetResponse contains the information stored about a given match
+type MatchGetResponse struct {
+	ID        int
+	userOne   int
+	userTwo   int
+	matchedOn string
+}
+
+// MatchListResponse contains the information stored about all matches
+type MatchListResponse struct {
+	IDs []int
+}
+
+// MatchCreateRequest contains the information required to create a new match
+type MatchCreateRequest struct {
+	ID1 string `valid:"type(string),required,stringlength(1|255)"`
+	ID2 string `valid:"type(string),required,stringlength(1|255)"`
+}
+
+// MatchCreateResponse contains the information stored about the newly created match
+type MatchCreateResponse struct {
+	ID        int
+	userOne   int
+	userTwo   int
+	matchedOn string
+}
+
+type MatchUpdateRequest struct {
+	ID      int
+	userOne int
+	userTwo int
+}
+
+func executeQueryWithResponses(query string, args ...interface{}) (*sql.Rows, error) {
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Thoughts on this, to make the error handling more clearly bound to the source?

```go
if db, err := sql.Open("postgres", connStr); err != nil {
	return nil, err
}
```

Reply:
Not a particularly strong feeling, but prefer as is